moto/firehose/urls.py

```diff
@@ -0,0 +1,6 @@
+"""Firehose base URL and path."""
+from .responses import FirehoseResponse
+
+
+url_bases = ["https?://firehose.(.+).amazonaws.com"]
```

Comment:
```suggestion
url_bases = ["https?://firehose\.(.+)\.amazonaws.com"]
```
This regex was wrong for all other services as well - we should ensure it matches a literal `.`, otherwise it will match any character. This should fix the test_bucketname_starting_with_service_name[firehose-]-test that's currently failing

Reply:
I wondered about that, but when I checked other services, some escaped the dots, some didn't.  I'll fix the scaffold tool as well.  But why does it work for the other services and fail for firehose?

Reply:
I submitted a PR some time ago that was supposed to fix all of them - do you remember which services do not escape it yet?

Reply:
There's a long list:

```
cd moto/moto
find . -name "urls.py"  | xargs grep "(.+).amazonaws"
```